File: haskell/leap/src/LeapYear.hs
module LeapYear (isLeapYear) where

isLeapYear :: Int -> Bool
isLeapYear year =
	if (isDivisibleBy 4 year) then
		if (isDivisibleBy 100 year) then
			isDivisibleBy 400 year
		else
			True
	else
		False

isDivisibleBy divisor n =
	rem n divisor == 0

File: fsharp/leap/LeapYear.fs
module LeapYear

let isLeapYear year =
    let by4 = year % 4 = 0
    let by100 = year % 100 = 0
    let by400 = year % 400 = 0
    match year with
    | _ when by4 && by100 -> by400
    | _ when by4 -> true
    | _ -> false